numba/typeinfer.py

```diff
@@ -255,6 +256,31 @@ class BuildSetConstraint(_BuildContainerConstraint):
     container_type = types.Set
 
 
+class BuildMapConstraint(object):
+
+    def __init__(self, target, items, loc):
+        self.target = target
+        self.items = items
+        self.loc = loc
+
+    def __call__(self, typeinfer):
+        with new_error_context("typing of dict at {0}", self.loc):
+            typevars = typeinfer.typevars
+            tsets = [(typevars[k.name].getone(), typevars[v.name].getone())
+                     for k, v in self.items]
+            if not tsets:
+                typeinfer.add_type(self.target,
+                                   types.DictType(types.undefined,
+                                                  types.undefined),
+                                   loc=self.loc)
+            else:
+                head = tsets[0]
```

Comment:
is the assignment needed here? `key_type, value_type = tsets[0]`?

Reply:
good catch